company-meghanada.el

```diff
@@ -63,8 +63,8 @@
   (company-mode t)
   (set (make-local-variable 'company-backends) nil)
   (set (make-local-variable 'company-transformers) nil)
-  (set (make-local-variable 'company-idle-delay) 0)
-  (set (make-local-variable 'company-minimum-prefix-length) company-meghanada-prefix-length)
+  (when company-meghanada-prefix-length
+    (set (make-local-variable 'company-minimum-prefix-length) company-meghanada-prefix-length))
   (setq company-meghanada-trigger-regex (format company-meghanada--trigger company-meghanada-prefix-length company-meghanada-prefix-length))
```

Comment:
Then `company-minimum-prefix-value` will be the default value, out whatever value the user configured for their company settings, it won't be buffer local, so it'll use the globally configured value